extensions/ql-vscode/src/variant-analysis/markdown-generation.ts

```diff
@@ -42,7 +42,7 @@ interface VariantAnalysisMarkdown {
  * Generates markdown files with variant analysis results.
  */
 export async function generateVariantAnalysisMarkdown(
-  variantAnalysis: Pick<VariantAnalysis, "query">,
+  variantAnalysis: Pick<VariantAnalysis, "language" | "query">,
```

Comment:
Not a problem necessarily introduced in this PR, but I would prefer to _not_ use `Pick` if we're going to be getting more than 1 item, as it gradually becomes silly (e.g. https://github.com/github/vscode-codeql/pull/3287/files#diff-ba6fe1916e2a1bce3900880797ea49539e39f9f76c431c2742a4ca5926643b26R50)

Reply:
For this PR I'll just keep it simple and add the extra field. I don't feel this PR crosses any new silliness-thresholds that weren't already crossed.

I'm not really sure why we're using `Pick` here (or the place you linked). I'm not sure what benefit it gives us in terms of ease of writing or maintainability, over taking the whole object. Perhaps it makes craft data in the tests easier, but we have `createMockX` methods for that generally.